internal/controller/controlplane/k3kcontrolplane_controller.go

```diff
@@ -121,12 +159,12 @@ func (r *K3kControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	if cluster == nil {
 		// capi cluster owner may not be set immediately, but we don't want to process the cluster until it is
 		log.Info("K3kControlPlane did not have a capi cluster owner", "controlPlane", k3kControlPlane.Name)
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
10 seconds seems long time to requeue, is there a reason to wait that long?

Reply:
I'll return an error instead, letting the controller handle the exponential backoff. :+1: